### DISCUSSION TITLE
Using .net appsettings.json to create StreamConfig

### DISCUSSION THREAD

**Discussion by ppilev** [LGouellec/streamiz#264] [General]
hi @LGouellec,

I'm trying to initialize `StreamConfig` out of _appsettings.json_ file using `IOptions<>` pattern, however because `StreamConfig` is derived from `Dictionary<string, dynamic>` the property setters of `StreamConfig` are ignored and the json configuration is directly translated to key-value pairs into the base class dictionary.
Then I tried to modify my json config like `"auto.offset.reset": "Latest"` but I realized that this will just create an entry in the dictionary with a string value "Latest" and the type conversion won't be handled properly.

Ultimately I'd like to be able to reuse existing configurations I have for `ClientConfig`, `ConsumerConfig` and `ProducerConfig` and have them applied to `StreamConfig`, however the internal implementation of the later completely prevents it.

Do you see any way to workaround that issue?

Regards,

**Comment by LGouellec** [comment]
Hi @ppilev , 

I refactored the stream configuration implementation [here](https://github.com/LGouellec/kafka-streams-dotnet/issues/144). The issue is not enough documented.

A new method will expose `config.AddConfig(key, value);` and this method will 

But basically, in the 1.5 release, you could easily do this kind of thing 

``` csharp
            var streamConfig = new StreamConfig();
            
            streamConfig.AddConfig("application.id", "test-app");
            streamConfig.AddConfig("metrics.interval.ms", 100000);
            streamConfig.AddConfig("follow.metadata", true);
            streamConfig.AddConfig("bootstrap.servers", "localhost:9092");
            streamConfig.AddConfig("auto.offset.reset", AutoOffsetReset.Latest);
            streamConfig.AddConfig("acks", Acks.Leader);
            streamConfig.AddConfig("sasl.mechanism", SaslMechanism.ScramSha512);
```

But your point still raise an error if you call something like that. I think I can adapt my PR to manage this kind of feature.
``` csharp
            streamConfig.AddConfig("auto.offset.reset", "Latest");
```

**Reply by ppilev** [reply]
I've submitted this PR to describe my issue better
https://github.com/LGouellec/kafka-streams-dotnet/pull/265

regards,

**Reply by ppilev** [reply]
hey @LGouellec 

were you able to review it by any chance?

regards,

**Reply by LGouellec** [reply]
Hi @ppilev ,

It's in my todo list. I Will review it next week I hope.
